### PR TITLE
[H] Undeprecate BigInteger support, but document what it actually does.

### PR DESCRIPTION
**Pull request by Hixie**
Also, some code cleanup.

### CODE REVIEW

**Comment by Hixie** [comment]
See also https://github.com/flutter/engine/pull/6903

**Comment by Hixie** [comment]
cc @amirh

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 1]
```diff
@@ -209,14 +209,20 @@ class JSONMethodCodec implements MethodCodec {
...
 /// [readValueOfType].
 class StandardMessageCodec implements MessageCodec<dynamic> {
   /// Creates a [MessageCodec] using the Flutter standard binary encoding.
   const StandardMessageCodec();
 
-  // The codec serializes messages as outlined below. This format must
-  // match the Android and iOS counterparts.
+  // The codec serializes messages as outlined below. This format must match the
+  // Android and iOS counterparts and cannot change (as it's possible for
+  // someone to end up using this for persistent storage).
```
Comment (amirh) [inline packages/flutter/lib/src/services/message_codecs.dart]: Are we recommending to use it for persistent storage (or are we even ok with it?).
Just thinking that if not we probably shouldn't make any commitment of forward compatibility here...

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 2]
```diff
@@ -293,12 +302,36 @@ class StandardMessageCodec implements MessageCodec<dynamic> {
...
+  ///  * Map = 13
+  ///  * Reserved for future expansion: 14..127
   ///
   /// The codec can be extended by overriding this method, calling super
   /// for values that the extension does not handle. Type discriminators
   /// used by extensions must be greater than or equal to 128 in order to avoid
   /// clashes with any later extensions to the base class.
+  ///
+  /// The "larger integers" type, 5, is never used by [writeValue]. A subclass
+  /// could represent big integers from another package using that type. The
```
Comment (amirh) [inline packages/flutter/lib/src/services/message_codecs.dart]: Do you have some historical context as for whether we can just deprecate it? (the comment on the decoder for this mentions we are planning to deprecate it)

I just find it a little weird to have us specify the format, keep a decoder implementation but say if you want to encode subclass and implement exactly this format...

Comment (amirh) [reply packages/flutter/lib/src/services/message_codecs.dart]: Coming back here after seeing the readValueOfType change which is backwards incompatible anyways... why not just drop support for this type?

File: packages/flutter/lib/src/services/message_codecs.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/services/message_codecs.dart]: Doesn't matter what commitment we make. People will use it that way anyway.

File: packages/flutter/lib/src/services/message_codecs.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/services/message_codecs.dart]: We said we'd remove it a year ago. We didn't. We missed the boat on deprecation. This is just an attempt to rescue a clean API out of the ashes that remain.

**Review by amirh — APPROVED** [review]
LGTM

**Pull request merged**